src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx

```diff
@@ -25,78 +25,311 @@ import codeLevelMetrics from 'images/code-level-metrics.png'
 
-CodeStream's code-level metrics give you detailed insight into how your code is performing.
+The telemetry data available in CodeStream is contextual, meaning that it's all
+related to the code you have open in your IDE. In order to do this CodeStream
+needs to know what entities on New Relic are built by the repositories you have
+open, and there are a number of ways to make these associations.
+
+There are instances where CodeStream will prompt you to make an association if
+one doesn't exist. For example, if you try view an error from New Relic in
+CodeStream, and the entity the error belongs to isn't associated with a
+repository, CodeStream will ask you to identify the appropriate repository. The
+Observability section of the the CodeStream pane will also prompt you take make
+associations for all of the repositories you have open in your IDE.
 
 <img
-  title="A Python method with average duration, throughput, and error rate metrics."
-  alt="A screenshot of a Python method with average duration, throughput, and error rate telemetry."
-  src={codeLevelMetrics}
+  title="Associate repositories in your IDE with entities."
+  alt="A screenshot of associating repositories in your IDE with entities."
+  src={associateRepos}
 />
 
-<figcaption>When relevant, CodeStream's code-level metrics are shown in text directly above your Python methods, called a CodeLens.</figcaption>
+Making these associations on the fly when prompted is a great way to get
+started, however we recommend one of the following methods since they require
+less ongoing manual effort and eliminate the possibility of end-user mistakes,
+such as misconfigured remote URLs. These methods have the additional benefit of
+alllowing you to associate a repository with multiple entities. For example you
+might have different entities that represent different environments (e.g.,
+production, staging) and they may all be associated with the same repository.
+
+<CollapserGroup>
+  <Collapser
+    className="freq-link"
+    id="env-var"
+    title="Use environment variable with APM (recommended)"
+  >
+    Set the environment variable `NEW_RELIC_METADATA_REPOSITORY_URL`. New Relic APM agents create the repository entity and associate it with your application entity automatically.
+
+    This requires the SSH or HTTPS remote URL format. We recommend that this be set as part of your build pipeline.
+  </Collapser>
+
+  <Collapser
+    className="freq-link"
+    id="repo-ui"
+    title="Use the UI"
+  >
+    Once you've started sending data to New Relic, use the UI to connect your related repository. Go to the APM Summary page via **[one.newrelic.com](https://one.newrelic.com) > Explorer > Services - APM > (select an app)**, then look for the **Repositories** section at the bottom-right.
+
+    <img
+      title="The connect repository button."
+      alt="A screenshot of the connect repository button."
+      src={connectRepo1}
```

Comment:
I'm not seeing a `connectRepo1` import at the top, which might be another issue @planteater